### PR TITLE
docs: SOAP OpenAPI specification

### PR DESCRIPTION
**Pull request by Woynert**
**What's added:**
- [OpenAPI specification added](https://github.com/hawks-atlanta/gateway-java/blob/c6a06c5731500d727fb6f713761341a48f27669f/docs/spec.openapi.yml).
- Standardize and update soap type names.
- Updated `CLI.md` and `README.md`
- Remove RMI sample code from App.java

## `docs/spec.openapi.yml`

Describes requests exposed in the gateway SOAP service. Despite this service not being a REST API this specification was created for developers convenience, so that they don't have to deal with the WSDL file.

How to read:

- Endpoints → SOAP requests
- Response codes → ResStatus code field

### CODE REVIEW

File: docs/spec.openapi.yml [hunk 1]
```diff
@@ -0,0 +1,658 @@
...
+                $ref: '#/components/schemas/ResSession'
+        'default':
+          description: An error occurred, see other codes at [Authentication API Spec](https://github.com/hawks-atlanta/authentication-go/blob/main/docs/spec.openapi.yaml)
+          content:
+            Object:
+              schema:
+                $ref: '#/components/schemas/ResStatus'
+
+                
+  /auth_refresh:
```
Comment (PedroChaparro) [inline docs/spec.openapi.yml:54]: I think there is no `/refresh` endpoint in the auth micro-service

File: docs/spec.openapi.yml [hunk 2]
```diff
@@ -0,0 +1,658 @@
...
+              schema:
+                $ref: "#/components/schemas/ResStatus"
+        'default':
+          description: An error occurred. Returns error code and message. See possible responses at [Metadata API Spec endpoint `POST /files/share/{user_uuid}/{file_uuid}`](https://github.com/hawks-atlanta/metadata-scala/blob/main/docs/spec.openapi.yaml)
+          content:
+            Object:
+              schema:
+                $ref: '#/components/schemas/ResStatus'
+
+  /share_remove:
```
Comment (PedroChaparro) [inline docs/spec.openapi.yml:339]: Maybe`unshare_file` its more clear

File: docs/spec.openapi.yml [hunk 3]
```diff
@@ -0,0 +1,658 @@
...
+              schema:
+                $ref: '#/components/schemas/ResStatus'
+        'default':
+          description: An error occurred, see other codes at [Authentication API Spec](https://github.com/hawks-atlanta/authentication-go/blob/main/docs/spec.openapi.yaml)
+          content:
+            Object:
+              schema:
+                $ref: '#/components/schemas/ResStatus'
+
+  # file system
```
Comment (PedroChaparro) [inline docs/spec.openapi.yml:132]: The `/rename` endpoint is missing according to the [Metadata spec](https://github.com/hawks-atlanta/metadata-scala/blob/dev/docs/spec.openapi.yaml)

**Review by PedroChaparro — CHANGES_REQUESTED** [review]
Good work. It was a good idea to add links to the others specifications.

File: docs/spec.openapi.yml (hunk 1 above)
Comment (Woynert) [reply docs/spec.openapi.yml:54]: The `/challenge` endpoint effectively acts as a `/refresh`. It returns a new token.

File: docs/spec.openapi.yml (hunk 2 above)
Comment (Woynert) [reply docs/spec.openapi.yml]: You're right, 'just trying to make them all start with `share` 😅️

**Review by PedroChaparro — APPROVED** [review]

**Review by shoriwe — APPROVED** [review]
😱🚀🚀🚀

**Pull request merged**